python/lsst/ts/electrometer/commander.py

```diff
@@ -119,21 +129,33 @@ async def send_command(
         -------
         reply
         """
-
+        if timeout is None: 
+            self.log.debug(f"Will use timeout {self.timeout}s")
+        else:
+            self.log.debug(f"Will use timeout {timeout}s")
         async with self.lock:
             msg = msg + self.command_terminator
             msg = msg.encode("ascii")
             if self.writer is not None:
                 self.log.debug(f"Commanding using: {msg}")
                 self.writer.write(msg)
                 await self.writer.drain()
+                if self.brand == "Keysight":
+                    # flush the echo
+                    _ = await asyncio.wait_for(
+                        self.reader.readuntil(self.reply_terminator),
+                        timeout=self.timeout
+                    )
```

Comment:
Not sure why this is necessary as the line below contains the same info. With the Keysight you want to send this command whether or not `has_reply`?

Reply:
With the Keysight the first reply is always just a repeat of the command we sent (an echo, if you will). This receives and then throws that echo away. The real response is then in the line below. I think the Keithley doesn't send an echo back so doesn't need this extra read.

Reply:
Ah interesting. Will you add a comment to the line there so we remember that in the future?